contracts/prize-strategy/PrizeStrategy.sol

```diff
@@ -127,49 +133,134 @@ contract PrizeStrategy is PrizeStrategyStorage,
+  /// @param token The token they are withdrawing (i.e. sponsorship or ticket)
+  /// @return The additional fee the the user needs to pay.  Credit is taken into account.
+  function calculateInstantWithdrawalFee(address from, uint256 amount, address token) public override returns (uint256) {
+    return _calculateInstantWithdrawalFee(from, amount, token);
+  }
+
+  /// @notice Calculates the instant withdrawal fee for a user, and burns the credit consumed.
+  /// @param from The user who is withdrawing
+  /// @param amount The amount of collateral they are withdrawing
+  /// @param token The token they are withdrawing (i.e. sponsorship or ticket)
+  /// @return The additional fee the the user needs to pay.  Credit is taken into account.
+  function _calculateInstantWithdrawalFee(address from, uint256 amount, address token) internal returns (uint256) {
     if (token == address(ticket)) {
-      uint256 totalFee = _calculateExitFee(amount);
-      uint256 totalCredit = _balanceOfTicketCredit(from);
-      uint256 burnAmount;
-      uint256 fee;
-      if (totalCredit >= totalFee) {
-        burnAmount = totalFee;
-      } else {
-        burnAmount = totalCredit;
-        fee = totalFee.sub(totalCredit);
-      }
       _accrueTicketCredit(from, ticket.balanceOf(from));
-      if (burnAmount > 0) {
-        creditBalances[from].credit = uint256(creditBalances[from].credit).sub(burnAmount).toUint128();
+      uint256 totalFee = _calculateExitFeeWithValues(amount, previousPrizeAverageTickets, previousPrize);
+      uint256 feeCredit = calculateFeeCredit(from, totalFee);
+      uint256 actualFee = totalFee.sub(feeCredit);
+      if (feeCredit > 0) {
+        creditBalances[from].credit = uint256(creditBalances[from].credit).sub(feeCredit).toUint128();
       }
-      return fee;
+      return actualFee;
     }
     return 0;
   }
 
-  function calculateWithdrawalUnlockTimestamp(address, uint256, address token) external override returns (uint256) {
+  /// @notice Calculates how much of a fee can be covered by the user's credit
+  /// @param user The user whose credit should be checked
+  /// @param fee The fee we are trying to cover
+  /// @return The amount of the fee that credit can cover
+  function calculateFeeCredit(address user, uint256 fee) internal view returns (uint256) {
+    uint256 credit = uint256(creditBalances[user].credit);
+    uint256 feeCredit;
+    if (credit >= fee) {
+      feeCredit = fee;
+    } else {
+      feeCredit = credit;
+    }
+    return feeCredit;
+  }
+
+  /// @notice Calculates the withdrawal unlock timestamp.  This function also accrues their ticket credit.
+  /// @param user The user who wishes to withdraw
+  /// @param token The token they are withdrawing
+  /// @return The absolute timestamp after which they are allowed to withdraw
+  function calculateWithdrawalUnlockTimestamp(address user, uint256 amount, address token) external override returns (uint256) {
```

Comment:
We could add the variable name to the return statement `returns (uint256 timelock)` and then potentially save 2 lines of code in the body, as well as providing named return params to the caller.